webapp/graphite/render/evaluator.py

```diff
@@ -91,13 +102,39 @@ def evaluateTokens(requestContext, tokens, replacements=None, pipedArg=None):
     kwargs = dict([(kwarg.argname, evaluateTokens(requestContext, kwarg.args[0], replacements))
                    for kwarg in tokens.call.kwargs])
 
+    def handleInvalidParameters(e):
+      if not getattr(handleInvalidParameters, 'alreadyLogged', False):
```

Comment:
Is the intent here that this would only log a single error per string that's being evaluated?

Reply:
If the enforcement of validation is disabled, it can happen that we end up in `handleInvalidParameters()` twice, if first the call to the validator raises an `InputParameterError` and later the call the to function raises another one. For example if an invalid aggregator string has been passed to `aggregate()` this could happen. In these cases we don't need to log two error messages about the same issue.

Reply:
Ok, that makes sense I think